dbt/adapters/databricks/impl.py

```diff
@@ -185,3 +210,19 @@ def _get_columns_for_catalog(self, relation: DatabricksRelation) -> Iterable[Dic
             as_dict["column_name"] = as_dict.pop("column", None)
             as_dict["column_type"] = as_dict.pop("dtype")
             yield as_dict
+
+    @contextmanager
+    def _catalog(self, catalog: Optional[str]) -> Iterator[None]:
+        current_catalog: Optional[str] = None
+        try:
+            if catalog is not None:
+                current_catalog = self.execute_macro(CURRENT_CATALOG_MACRO_NAME)[0][0]
```

Comment:
Just curious is the result of current_catalog cached somewhere?

Reply:
No, it's not cached.